expat/tests/common.c

```diff
@@ -185,14 +185,16 @@ _XML_Parse_SINGLE_BYTES(XML_Parser parser, const char *s, int len,
   if (chunksize > 0) {
     // parse in chunks of `chunksize` bytes as long as possible
     for (; offset + chunksize < len; offset += chunksize) {
-      enum XML_Status res = XML_Parse(parser, s + offset, chunksize, XML_FALSE);
+      const char *const start = (s == NULL) ? NULL : (s + offset);
+      enum XML_Status res = XML_Parse(parser, start, chunksize, XML_FALSE);
```

Comment:
It might be simpler to just move s forward after each parse: `s += chunksize`.

Reply:
@Snild-Sony addressed in more detail at https://github.com/libexpat/libexpat/commit/7f0120dacbd08cacd6ce2e94247f2763eda78790#r131929667 , adjusted.